lib/vm/src/backends/singlepass.rs

```diff
@@ -40,5 +40,10 @@ pub fn set_gas(instance: &mut Instance, limit: u64) {
 
 pub fn get_gas(instance: &Instance) -> u64 {
     let used = metering::get_points_used(instance);
-    GAS_LIMIT - used
+    // when running out of gas, get_points_used can exceed GAS_LIMIT
+    if used < GAS_LIMIT {
```

Comment:
Safer for sure

Reply:
Without that, I got overflow crashes

Reply:
I did also months ago with huge values, but then just used smaller numbers and forgot about it. Good fix